File: week2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

float get_characters(string text)
{
    int cnt = 0;
    int i = -1;

    while (text[++i])
    {
        if (isalpha(text[i]))
        {
            cnt++;
        }
    }
    return (cnt);
}

float get_words(string text)
{
    int cnt = 1;
    int i = -1;

    while (text[++i])
    {
        if (text[i] == ' ')
        {
            cnt++;
        }
    }
    return (cnt);
}

float get_sentences(string text)
{
    int cnt = 0;
    int i = -1;

    while (text[++i])
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            cnt++;
        }
    }
    return (cnt);
}

int main(void)
{
    string text = get_string("Text: ");
    float characters = get_characters(text);
    float words = get_words(text);
    float sentences = get_sentences(text);
    float L = characters / words * 100;
    float S = sentences / words * 100;
    //printf("characters: %f words: %f sentences: %f L: %f S: %f", characters, words, sentences, L, S);
    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", (int)round(index));
    }
    return (0);
}

File: week6/readability/readability.py
from cs50 import get_string
from math import floor

alphabet = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
space = " "
end = ".!?"
text = get_string("Text: ")
characters = 0
words = 1
sentences = 0
for s in alphabet:
    characters += text.count(s)
for s in space:
    words += text.count(s)
for s in end:
    sentences += text.count(s)
L = characters / words * 100
S = sentences / words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print("Grade " + str(index))
